### PR TITLE
chore: add create project events

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #3332 

### Description:

# New and updated FE events:
`create_project_manually_click.click` 
`copy_create_project_code_click.click`
`documentation_button.click` (event already existed) with property `action` and value `getting_started`

# Updated BE events:
`personal_access_token.created` now has properties `userId`, `autoGenerated` and `method` 
`project.created` and `project.updated`  now has property `method` ( we also fixed the isPreview value on create )

## Notes:
Leveraged request headers so the BE can know if the request was made from the CLI or the WEB_APP or somewhere else. These headers are applied to all requests. 
Values: 'CLI', 'WEB_APP' and 'UNKNOWN'

### CODE REVIEW

File: packages/frontend/src/pages/CreateProject.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ const CreateProject: FC = () => {
         mutateAccessToken({
             expiresAt,
             description: `Generated by the Lightdash UI for CLI at ${generatedAtString}`,
+            autoGenerated: true,
```
Comment (IrakliJani) [inline packages/frontend/src/pages/CreateProject.tsx:56]: 👍🏻

**Review by IrakliJani — APPROVED** [review]

File: packages/common/src/utils/api.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { isRequestMethod, RequestMethod } from '../types/api';
+
+export const LightdashRequestMethodHeader = 'Lightdash-Request-Method';
```
Comment (rephus) [inline packages/common/src/utils/api.ts:3]: :kissing: :ok_hand:

File: packages/common/src/utils/api.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+export const getRequestMethod = (
+    headerValue: string | undefined,
+): RequestMethod =>
+    isRequestMethod(headerValue) ? headerValue : RequestMethod.UNKNOWN;
```
Comment (rephus) [inline packages/common/src/utils/api.ts:8]: nice catch, some people using an outdated cli will not be sending this header

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 4]
```diff
@@ -152,7 +154,8 @@ export class ProjectService {
                 warehouseConnectionType: data.warehouseConnection.type,
                 organizationId: user.organizationUuid,
                 dbtConnectionType: data.dbtConnection.type,
-                isPreview: true,
+                isPreview: data.type === ProjectType.PREVIEW,
```
Comment (rephus) [inline packages/backend/src/services/ProjectService/ProjectService.ts:157]: :+1:

**Review by rephus — APPROVED** [review]
LGMT, nice new header

**Pull request merged**